test/helpers/fixture_deploy_helper.rb

```diff
@@ -183,22 +171,29 @@ def build_kubectl(log_failure_by_default: true, timeout: '5s')
       log_failure_by_default: log_failure_by_default, default_timeout: timeout)
   end
 
-  def namespace_globals(fixtures, selector)
-    selector_key, selector_value = selector.split("=")
+  def add_unique_prefix_for_test(original_name)
+    "t#{Digest::MD5.hexdigest(@namespace)}-#{original_name}"
+  end
+
+  def apply_scope_to_resources(fixtures, labels:)
+    labels = Krane::LabelSelector.parse(labels).to_h
     fixtures.each do |_, kinds_map|
       kinds_map.each do |_, resources|
         resources.each do |resource|
-          resource["metadata"]["name"] = (resource["metadata"]["name"] + Digest::MD5.hexdigest(@namespace))[0..63]
-          resource["metadata"]["labels"] ||= {}
-          resource["metadata"]["labels"][selector_key] = selector_value
+          resource["metadata"]["labels"] = (resource.dig("metadata", "labels") || {}).merge(labels) if labels.present?
+          if resource["kind"] == "CustomResourceDefinition"
+            %w(kind listKind plural singular).each do |field|
+              if (original_name = resource.dig("spec", "names", field))
+                resource["spec"]["names"][field] = add_unique_prefix_for_test(original_name)
+              end
+            end
+            # metadata.name has to be composed this way for CRDs
+            resource["metadata"]["name"] = "#{resource['spec']['names']['plural']}.#{resource['spec']['group']}"
+          else
+            resource["metadata"]["name"] = add_unique_prefix_for_test(resource["metadata"]["name"])[0..63]
```

Comment:
names can be 253 chars long https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names. Its only labels that are limited to 63 https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set